shared/test/test_net_sim_api.rb

```diff
@@ -71,6 +71,54 @@ def test_create_read_update_delete
     assert read_records.first.nil?, 'Table was not empty'
   end
 
+  def test_create_multiple_records
+    created_ids = []
+    # Sending any number of records as an array should result in an
+    # array being returned
+    record_create_response = create_record([{name: 'alice', age: 7, male: false}])
+    assert record_create_response.is_a?(Array)
+    assert_equal 1, record_create_response.length
+    assert_equal 1, read_records().length
+    created_ids.push(record_create_response[0]['id'])
+
+    # Sending a record as a hash should result in a hash being returned
+    record_create_response = create_record({name: 'fred', age: 12, male: true})
+    assert record_create_response.is_a?(Hash)
+    assert_equal 2, read_records().length
+    created_ids.push(record_create_response['id'])
+
+    # Sending several records should result in them all being inserted
+    record_create_response = create_record([
+      {name: 'nancy', age: 9, male: false},
+      {name: 'drew', age: 11, male: true}
+    ])
+    assert record_create_response.is_a?(Array)
+    assert_equal 2, record_create_response.length
+    assert_equal 4, read_records().length
+    created_ids.push(record_create_response[0]['id'])
+    created_ids.push(record_create_response[1]['id'])
+
+    # sending an empty array should be a no-op
+    record_create_response = create_record([])
+    assert record_create_response.is_a?(Array)
+    assert_equal 0, record_create_response.length
+    assert_equal 4, read_records().length
+
+    # sending a value that is neither an array nor a hash should fail
+    record_create_response = create_record(1)
+    assert_equal 400, @net_sim_api.last_response.status
+    assert_equal 4, read_records().length
+
+    # sending an array containing a value that is neither an array nor a
+    # hash should fail
+    record_create_response = create_record([1])
```

Comment:
Lint/UselessAssignment: Useless assignment to variable - `record_create_response`.